spyder/widgets/panels/scrollflag.py

```diff
@@ -34,98 +56,154 @@ def paintEvent(self, event):
-                    # Breakpoints
+                    # Paint the breakpoints
                     self.set_painter(painter, self.editor.breakpoint_color)
-                    painter.drawRect(make_flag(position))
+                    painter.drawRect(make_flag(line_number))
             block = block.next()
 
-        # Occurrences
+        # Paint the occurrences
         if self.editor.occurrences:
             self.set_painter(painter, self.editor.occurrence_color)
             for line_number in self.editor.occurrences:
-                position = self.value_to_position(line_number)
-                painter.drawRect(make_flag(position))
+                painter.drawRect(make_flag(line_number))
 
-        # Found results
+        # Paint the found results
         if self.editor.found_results:
             self.set_painter(painter, self.editor.found_results_color)
             for line_number in self.editor.found_results:
-                position = self.value_to_position(line_number)
-                painter.drawRect(make_flag(position))
-
-        # Painting the slider range
-        pen_color = QColor(Qt.white)
-        pen_color.setAlphaF(.8)
-        painter.setPen(pen_color)
-        brush_color = QColor(Qt.white)
-        brush_color.setAlphaF(.5)
-        painter.setBrush(QBrush(brush_color))
-        painter.drawRect(make_slider(self.editor.firstVisibleBlock().blockNumber()))
+                painter.drawRect(make_flag(line_number))
+
+        # Paint the slider range
+        alt = QApplication.queryKeyboardModifiers() & Qt.AltModifier
+        cursor_pos = self.mapFromGlobal(QCursor().pos())
+        if ((self.rect().contains(cursor_pos) or alt) and self.slider):
+            pen_color = QColor(Qt.gray)
+            pen_color.setAlphaF(.85)
+            painter.setPen(pen_color)
+            brush_color = QColor(Qt.gray)
+            brush_color.setAlphaF(.5)
+            painter.setBrush(QBrush(brush_color))
+            painter.drawRect(self.make_slider_range(cursor_pos))
+
+    def enterEvent(self, event):
+        """Override Qt method"""
+        self.update()
+
+    def leaveEvent(self, event):
+        """Override Qt method"""
+        self.update()
+
+    def mouseMoveEvent(self, event):
+        """Override Qt method"""
+        self.update()
 
     def mousePressEvent(self, event):
         """Override Qt method"""
         vsb = self.editor.verticalScrollBar()
-        value = self.position_to_value(event.pos().y()-1)
-        vsb.setValue(value-.5*vsb.pageStep())
+        value = self.position_to_value(event.pos().y())
+        vsb.setValue(value-vsb.pageStep()/2)
 
-    def get_scale_factor(self, slider=False):
+    def get_scrollbar_position_height(self):
+        """Return the pixel span height of the scrollbar area in which
+        the slider handle may move"""
+        vsb = self.editor.verticalScrollBar()
+        style = vsb.style()
+        opt = QStyleOptionSlider()
+        vsb.initStyleOption(opt)
```

Comment:
This line is causing this error in my local configuration (pyqt5, python3):

```
Traceback (most recent call last):
  File "/home/rlaverde/code/spyder/spyder/widgets/panels/scrollflag.py", line 77, in paintEvent
    painter.drawRect(make_flag(line_number))
  File "/home/rlaverde/code/spyder/spyder/widgets/panels/scrollflag.py", line 168, in make_flag_qrect
    position = self.value_to_position(value+0.5)
  File "/home/rlaverde/code/spyder/spyder/widgets/panels/scrollflag.py", line 158, in value_to_position
    return (y-vsb.minimum())*self.get_scale_factor()+self.offset
  File "/home/rlaverde/code/spyder/spyder/widgets/panels/scrollflag.py", line 152, in get_scale_factor
    return (self.get_scrollbar_position_height() /
  File "/home/rlaverde/code/spyder/spyder/widgets/panels/scrollflag.py", line 136, in get_scrollbar_position_height
    vsb.initStyleOption(opt)
RuntimeError: no access to protected functions or signals for objects not created from Python
```

Reply:
I'll see how I fix that, thanks for the heads up. I do not get this error on Windows with pyqt5 and python3.

Reply:
Do you think this is why ci/cicleci is failing? If yes, that would help me debug that.

Reply:
Ok yeah, it's totally related. I don't understand why we get this error on Ubuntu and not on Windows. Thanks to the test on ci/cicleci, I'll be able to debug that.

Reply:
@rlaverde I think this should work now. It is working on ci/cicleci.

Reply:
yes, now I could test it 😄